### DISCUSSION TITLE
Successful RPM Build and Install (Fedora 40)

### DISCUSSION THREAD

**Discussion by guzman109** [usebruno/bruno#2372] [Q&A]
Does anyone have a way to build an RPM and install it?

**Comment by Zen3515** [comment]
I tested with version 1.21.0

The answer was made here #1368 

> Right now the build system is a bit confusing, I got the same error some time ago.
> 
> To build the RPM package first you need to build the NextJS app in `packages/bruno-app`, you can use the `build:web` NPM script in the root `package.json` for that.
> 
> I usually use this shell script:
> 
> ```shell
> git clone https://github.com/usebruno/bruno.git usebruno-monorepo
> cd usebruno-monorepo
> 
> # checkouts the latest stable version (at the time of writing).
> # you can skip this following command to build directly from the `main` branch
> git checkout v1.7.0
> 
> # ensure you are using the required Node version defined in the `.nvmrc` file
> # at the time of writing it is Node v18.13.0
> node --version
> 
> # installs dependencies from the lockfile, you can use `npm install` too
> npm ci
> 
> # these three following commands are documented in the `contributing.md` file
> npm run build:graphql-docs
> npm run build:bruno-query
> npm run build:bruno-common
> 
> # builds the NextJS app
> # the ENV environment variable:
> # prod - telemetry enabled, redux debug middleware disabled
> # dev  - telemetry disabled, redux debug middleware enabled
> ENV=prod npm run build:web
> 
> # builds the RPM package
> npm run build:electron:rpm
> 
> # installs the RPM package, you may need to use `sudo` here
> dnf install packages/bruno-electron/out/*.rpm
> ```
